mycli/packages/special/iocommands.py

```diff
@@ -240,3 +240,50 @@ def execute_system_command(arg, **_):
         return [(None, None, None, response)]
     except OSError as e:
         return [(None, None, None, 'OSError: %s' % e.strerror)]
+
+@export
+def set_redirected_output(append=False, filename=None, enabled=True):
+    global use_redirected_output
+    use_redirected_output = {'append': append, 'filename':filename, 'enabled': enabled}
+
+@export
+def get_redirected_output():
+    return use_redirected_output
+
+@export
+def redirectmatch(sql):
+    filename = None
+    if sql[-1] in ("'", '"'): # find filename in quoted redirect
+        quote = sql[-1]
+        m = re.match(r'.*([^\\]'+quote+r'.*?'+quote+'$)', sql)
```

Comment:
Hmm, actually I say that, but then I see that you use the length of the filename in line 267 to remove it from the query. That wouldn't work with my suggestion.